### PR TITLE
Port scenario_conditions to ROS2

### PR DESCRIPTION
**Pull request by nnmm**
Based on the excellent work of @jilaada.

### CODE REVIEW

File: condition/scenario_conditions/package.xml [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
 
+  <depend>rclcpp</depend>
   <depend>pluginlib</depend>
-  <depend>roscpp</depend>
   <depend>scenario_api</depend>
   <depend>scenario_intersection</depend>
   <depend>scenario_utility</depend>
   <depend>visualization_msgs</depend>
-  <depend>yaml-cpp</depend>
+  <depend>yaml_cpp_vendor</depend>
```
Comment (jilaada) [inline condition/scenario_conditions/package.xml:19]: I forgot to add the boost dependency here while I was doing the original port

File: condition/scenario_conditions/src/condition_visualizer.cpp [hunk 2]
```diff
@@ -4,17 +4,18 @@
 
 namespace scenario_conditions
 {
-ConditionVisualizer::ConditionVisualizer()
+ConditionVisualizer::ConditionVisualizer(const rclcpp::Node::SharedPtr node)
+: node_(node)
 {
-  auto pnh = ros::NodeHandle("~");
-  pub_marker_ = pnh.advertise<visualization_msgs::MarkerArray>("condition_marker", 1);
+  // auto pnh = ros::NodeHandle("~");
```
Comment (jilaada) [inline condition/scenario_conditions/src/condition_visualizer.cpp]: I think this can be removed

**Review by jilaada — COMMENTED** [review]
Quickly skimmed this as I remembered there were some things I may have forgotten to remove/change while porting. Thanks for finishing this package off!

File: condition/scenario_conditions/src/condition_visualizer.cpp (hunk 2 above)
Comment (nnmm) [reply condition/scenario_conditions/src/condition_visualizer.cpp]: Done :)

File: condition/scenario_conditions/package.xml (hunk 1 above)
Comment (nnmm) [reply condition/scenario_conditions/package.xml:19]: Done!

File: condition/scenario_conditions/CMakeLists.txt [hunk 3]
```diff
@@ -1,51 +1,28 @@
...
-)
+### Dependencies
+find_package(ament_cmake_auto REQUIRED)
+ament_auto_find_build_dependencies()
 
-include_directories(
-  include
-  ${catkin_INCLUDE_DIRS}
-  ${YAML_CPP_INCLUDE_DIR}
+set(SCENARIO_CONDITIONS_SRC
```
Comment (fred-apex-ai) [inline condition/scenario_conditions/CMakeLists.txt]: [nit] I would omit the variable and just add the file verbatim to the library below

File: condition/scenario_conditions/include/scenario_conditions/condition_visualizer.h [hunk 4]
```diff
@@ -11,14 +12,15 @@ class ConditionManager;
...
-  ConditionVisualizer();
+  ConditionVisualizer(const rclcpp::Node::SharedPtr node);
   void publishMarker(ConditionManager& manager);
 
 private:
   void addMarker(std::string name, bool result, bool is_success_condition);
 
-  ros::Publisher pub_marker_;
-  visualization_msgs::MarkerArray marker_array_;
+  rclcpp::Node::SharedPtr node_;
```
Comment (fred-apex-ai) [inline condition/scenario_conditions/include/scenario_conditions/condition_visualizer.h:21]: [nit] you could probably make this `const`, or do we want to change the node ever?

File: condition/scenario_conditions/package.xml (hunk 1 above)
Comment (fred-apex-ai) [reply condition/scenario_conditions/package.xml:19]: but doesn't that require `target_link_libraries()` call in `CMakeLists.txt` to link to boost in a correct manner, if boost istn' in the system paths the linker uses?

**Review by fred-apex-ai — APPROVED** [review]

File: condition/scenario_conditions/include/scenario_conditions/condition_visualizer.h (hunk 4 above)
Comment (nnmm) [reply condition/scenario_conditions/include/scenario_conditions/condition_visualizer.h:21]: None of the other members in this package are const even though they could be. I think it's better to stay consistent for now.

Comment (fred-apex-ai) [reply condition/scenario_conditions/include/scenario_conditions/condition_visualizer.h:21]: ok, let's just keep in mind for autoware.auto merges to look at const correctness

File: condition/scenario_conditions/CMakeLists.txt (hunk 3 above)
Comment (nnmm) [reply condition/scenario_conditions/CMakeLists.txt]: Done.

File: condition/scenario_conditions/package.xml (hunk 1 above)
Comment (nnmm) [reply condition/scenario_conditions/package.xml:19]: I simply removed the Boost dependency. See the stringstream above.

File: condition/scenario_conditions/include/scenario_conditions/condition_visualizer.h (hunk 4 above)
Comment (nnmm) [reply condition/scenario_conditions/include/scenario_conditions/condition_visualizer.h:21]: Agree

File: condition/scenario_conditions/include/scenario_conditions/condition_base.h [hunk 5]
```diff
@@ -15,8 +17,11 @@ class ConditionBase
 
   ConditionBase(const std::string & type)
     : type_ { type }
-    , name_ { type + "Condition<" + boost::lexical_cast<std::string>(this) + ">" }
-  {}
+  {
+    std::stringstream ss;
+    ss << type << "Condition<" << static_cast<const void*>(this) << ">";
```
Comment (fred-apex-ai) [inline condition/scenario_conditions/include/scenario_conditions/condition_base.h:22]: :+1: for removing the boost dep. This seems odd to save the pointer as a string in `name_`. But whatever..

**Review by fred-apex-ai — APPROVED** [review]

**Comment by fred-apex-ai** [comment]
@nnmm  @esteve I approved, so from  my side ready to merge.

**Comment by nnmm** [comment]
@fred-apex-ai Thanks, we're only waiting on https://github.com/tier4/scenario_runner.iv.universe/pull/26 being merged

**Comment by nnmm** [comment]
@esteve This one (and #20) are next in the dependency graph. CI should be green in an hour.

**Pull request merged**